dash/development/base_component.py

```diff
@@ -265,6 +312,64 @@ def __len__(self):
         return length
 
 
+def js_to_cerberus_type(type_object):
+    converters = {
+        'None': lambda x: {},
+        'func': lambda x: {},
+        'custom': lambda x: {},
+        'node': lambda x: {'type': 'component'},
+        'enum': lambda x: {
+            'anyof': [js_to_cerberus_type(v) for v in x['value']]
+        },
+        'union': lambda x: {
+            'anyof': [js_to_cerberus_type(v) for v in x['value']]
+        },
+        'any': lambda x: {
+            'anyof_type': ['bool', 'number', 'string', 'dict', 'list']
+        },
+        'string': lambda x: {'type': 'string'},
+        'bool': lambda x: {'type': 'boolean'},
+        'number': lambda x: {'type': 'number'},
+        'integer': lambda x: {'type': 'number'},
+        'object': lambda x: {'type': 'dict'},
+        'objectOf': lambda x: {
+            'type': 'dict',
+            'allow_unknown': False,
+            'schema': js_to_cerberus_type(x['value'])
+        },
+        'array': lambda x: {'type': 'list'},
+        'arrayOf': lambda x: {
+            'type': 'list',
+            'allow_unknown': False,
+            'schema': js_to_cerberus_type(x['value'])
```

Comment:
I'm assuming that this is recursive in a sense? That is, that this could contain it's own `arrayOf` which could be itself an `arrayOf`, etc?

Reply:
Yes, it seems to be working well with recursive definitions.